core/api/service/base_request.hpp

```diff
@@ -76,29 +67,46 @@ namespace kagome::api::chain::request {
     }
 
    private:
-    KAGOME_LOAD_VALUE(int32_t) {
-      if (!src.IsInteger32())
-        throw jsonrpc::InvalidParametersFault("invalid argument");
-
-      dst = src.AsInteger32();
+    template <typename T>
+    void loadValue(boost::optional<T> &dst, const jsonrpc::Value &src) {
+      if (!src.IsNil()) {
+        T t;
+        loadValue(t, src);
+        dst = std::move(t);
+      } else {
+        dst = boost::none;
+      }
     }
 
-    KAGOME_LOAD_VALUE(int64_t) {
-      if (!src.IsInteger64())
-        throw jsonrpc::InvalidParametersFault("invalid argument");
+    template <typename T, typename = std::enable_if<std::is_integral_v<T>>>
+    void loadValue(T &dst, const jsonrpc::Value &src) {
+      if (not src.IsInteger32() and not src.IsInteger64()) {
```

Comment:
It seems ```not src.IsInteger32()``` check is pointless. Because int32 also is an int64.

Reply:
```c++
if (!src.IsInteger64()) {
  ...
}

Reply:
No, it not right. Look at **.../json-lean/value.h:157**:
```
bool IsInteger32() const { return myType == Type::INTEGER_32; }
bool IsInteger64() const { return myType == Type::INTEGER_64; }
```
It is different types, although it is convertible.

Reply:
Wow! Calm down man, I can go blind!
How did they come up with this?

Reply:
Someweird is happening sometimes...